lib/r10k/api.rb

```diff
@@ -72,6 +77,42 @@ def parse_puppetfile(io_or_content)
       return { :modules => builder.build }
     end
 
+    # Create an unresolved environment hashmap representing the desired state of a Puppet environment based on a source and branch.
+    #
+    # @param source [Hash] A hashmap representing a control repo source (like would be defined in r10k.yaml)
+    # @param branch_name [String] Branch in the control repo to build an envmap for.
+    # @param base_cachedir [String] Root of where r10k is caching things.
+    # @return [Hash] A hashmap representing the desired state of the environment matching the given branch in the given source.
+    # @raise [NotImplementedError] Control repo source has a type that is not currently supported.
+    # @raise [RuntimeError] Something bad happened, see exception message.
+    def envmap_from_source(source, branch_name, base_cachedir)
+      # TODO: enforce valid Puppet environment name here?
+      env_name = branch_name
+      env_name = "#{source[:prefix]}_#{env_name}" if source[:prefix]
+
+      case source[:type].to_sym
+      when :git
```

Comment:
Why are we adding a source `type` class, converting it to a symbol, and then doing a comparison of that symbol rather than just checking the class type? Seems like `when R10K::Source::Git ...` will do the same thing with less work.